package-versions/grouping-and-overrides/multiple-groups.md

```diff
@@ -0,0 +1,120 @@
+        <PackageVersion Include="bar" Version="20.0.0" CentralManagementGroup="A" />
+    </ItemGroup>
+    <!-- Each <PackageVersion> within this <ItemGroup> belongs to the "B" group as specified in their CentralManagementGroup attributes. -->
+    <ItemGroup>
+        <PackageVersion Include="foo" Version="3.3.3" CentralManagementGroup="B" />
+        <PackageVersion Include="qux" Version="3.0.0" CentralManagementGroup="B" />
+        <!-- baz has a dependency on foo 303.0.0 -->
+        <!-- baz has a dependency on qux 303.0.0 -->
+        <!-- baz has a dependency on xyzzy 303.0.0 -->
+        <PackageVersion Include="baz" Version="30.0.0" CentralManagementGroup="B" />
+    </ItemGroup>
+</Project>
+```
+
+In this `Directory.Packages.props` file, there are three `CentralManagementGroup` of `<PackageVersion>` defined.
+
+1. The global `CentralManagementGroup` is automatically created, and all `<PackageVersion>` that do not have the `CentralManagementGroup` attribute defined belong to this group.
+2. `CentralManagementGroup="A"` defines a `CentralManagementGroup` with `<PackageVersion>` for 5 packages.
+3. `CentralManagementGroup="B"` defines a `CentralManagementGroup` with `<PackageVersion>` for 3 packages.
+
+*Example Package to Demonstrate Version Override*
+
+There are two `<PackageVersion>` defined for `Newtonsoft.Json` to demonstrate using both the global and a named `CentralManagementGroup`.
+
+- If a `<PackageReference>` does not specify a `CentralManagementGroup`, it will import version 12.0.2 from the global group.
+- If `CentralManagementGroup=A` is defined, it will import 11.0.2 from group A.
+
+### Sample Project 6: Multiple Named Groups
+
+#### 6.csproj
+
+```xml
+<Project>
+  <ItemGroup>
+    <!-- global group -->
+    <!-- Version 2.9.0 -->
+    <PackageReference Include="Serilog"/>
+    <!-- "A" group -->
+    <!-- Version 11.0.2 -->
+    <PackageReference Include="Newtonsoft.Json" CentralManagementGroup="A" />
+    <!-- "B" group -->
+    <!-- Version 3.3.3 -->
+    <PackageReference Include="foo" CentralManagementGroup="B" />
+  </ItemGroup>
+</Project>
+```
+
+This project uses three  `CentralManagementGroup`, `"A",` `"B"`, and the global group.
+
+All `<PackageReference>` dependencies are included between these `CentralManagementGroup` and are directly imported.
+
+### Sample Project 7: Multiple Named Groups with Transitive Dependencies
+
+#### 7.csproj
+
+```xml
+<Project>
+  <ItemGroup>
+    <!-- global group -->
+    <!-- Version 2.9.0 -->
+    <PackageReference Include="Serilog"/>
+    <!-- "A" group -->
+    <!-- Version 11.0.2 -->
+    <PackageReference Include="Newtonsoft.Json" CentralManagementGroup="A" />
+    <!-- "A" group -->
+    <!-- Version 20.0.0 -->
+    <PackageReference Include="bar" CentralManagementGroup="A" />
+    <!-- "B" group -->
+    <!-- Version 3.3.3 -->
+    <PackageReference Include="foo" CentralManagementGroup="B" />
+  </ItemGroup>
+</Project>
+```
```

Comment:
I would love to have something less noisy, and if possible relly on MsBuild `Property` by default in `fs/csproj`:
From:
```xml
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    <ItemGroup>
        <PackageReference Include="Newtonsoft.Json" CentralManagementGroup="A" />
        <PackageReference Include="foo" CentralManagementGroup="A" />
        <PackageReference Include="qux" CentralManagementGroup="A" />
        <PackageReference Include="xyzzy" CentralManagementGroup="A" />
        <PackageReference Include="bar" CentralManagementGroup="A" />
    </ItemGroup>
</Project>
```
to something like
```xml
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    <PropertyGroup>
        <CentralManagementGroup>A</CentralManagementGroup>
    </PropertyGroup>
    <ItemGroup>
        <PackageReference Include="Newtonsoft.Json" />
        <PackageReference Include="foo" />
        <PackageReference Include="qux" />
        <PackageReference Include="xyzzy" />
        <PackageReference Include="bar" />
    </ItemGroup>
</Project>
```

Reply:
FWIW if they _do_ go with the per-reference approach, you can still be less noisy with this syntax:

```xml
<Project>
    <ItemDefinitionGroup>
        <PackageReference>
            <CentralManagementGroup>A</CentralManagementGroup>
        </PackageReference>
    </ItemDefinitionGroup>
    <ItemGroup>
        <PackageReference Include="Newtonsoft.Json" />
        <PackageReference Include="foo" />
        <PackageReference Include="qux" />
        <PackageReference Include="xyzzy" />
        <PackageReference Include="bar" />
    </ItemGroup>
</Project>
```